src/snowflake/cli/api/project/schemas/entities/application_entity_model.py

```diff
@@ -39,3 +41,21 @@ class ApplicationEntityModel(EntityModelBase):
         title="Whether to enable debug mode when using a named stage to create an application object",
         default=None,
     )
+
+    @field_validator("identifier")
+    @classmethod
+    def append_test_resource_suffix_to_identifier(
+        cls, input_value: Identifier | str
+    ) -> Identifier | str:
+        identifier = (
+            input_value.name if isinstance(input_value, Identifier) else input_value
```

Comment:
So something like:

```
identifier = Identifier(input_value)  # accepts str or Identifier
return identifier.with_name(append_test_resource_suffix(identifier.name))  # returns a new instance with the name replaced
```

?

Reply:
I was thinking 2 separate validators (ordered). 
- It's just that conversion to Identifier might need to apply to all entity types (not only app and package)
- We have to think about templating as well - if users reference ctx.entities.x.identifier in case we modify the structure.

Reply:
oh, yes that would work too. As long as we don't have to copy the fields one by one.

Reply:
Updated using `return input_value.model_copy(update=dict(name=with_suffix))`